trunk/src/lsc/myloopdef.py

```diff
@@ -1809,37 +1811,42 @@ def get_list(epoch=None, _telescope='all', _filter='', _bad='', _name='', _id=''
         ll = ''
     return ll
 
-def get_standards(epoch, name, filters):
+def get_standards(epoch, name, filters, standard_name='all', match_by_site=False):
     epochs = process_epoch(epoch)
-    targetid = lsc.mysqldef.gettargetid(name, '', '', lsc.conn)
+
+    if standard_name == 'all':
+        std_join = 'JOIN targets AS targstd ON std.targetid = targstd.id'
+        is_standard = 'targstd.classificationid = 1'
+    else:
+        std_join = 'JOIN targetnames AS targstd ON std.targetid = targstd.targetid'
+        is_standard = 'targstd.name = "{}"'.format(standard_name)
+
+    if match_by_site:  # join on site, telescope class, and instrument type
+        tel_join = '''((photlco AS obj JOIN targetnames AS targobj ON obj.targetid = targobj.targetid)
+                                       JOIN telescopes AS telobj ON obj.telescopeid = telobj.id)
+                                       JOIN instruments AS instobj ON obj.instrumentid = instobj.id,
+                      ((photlco as std {})
+                                       JOIN telescopes AS telstd ON std.telescopeid = telstd.id)
+                                       JOIN instruments AS inststd ON std.instrumentid = inststd.id'''.format(std_join)
+        is_match = 'telobj.shortname = telstd.shortname AND instobj.type = inststd.type'
+    else:  # join on exact telescope and instrument
+        tel_join = '''photlco AS obj JOIN targetnames AS targobj ON obj.targetid = targobj.targetid,
+                      photlco AS std {}'''.format(std_join)
+        is_match = 'obj.telescopeid = std.telescopeid AND obj.instrumentid = std.instrumentid'
+
     query = '''SELECT DISTINCT std.filepath, std.filename, std.objname, std.filter,
```

Comment:
Does this query no longer hang if you try to query for one standard using another?

Reply:
No idea.

Reply:
@cmccully is this something you'd like @griffin-h to investigate before merging or can it be merged without it since we now avoid this situation to begin with?